examples/ogbn_train.py

```diff
@@ -0,0 +1,384 @@
+    def __init__(self, in_channels, hidden_channels, out_channels, num_layers,
+                 dropout, heads):
+        super().__init__()
+
+        self.num_layers = num_layers
+
+        self.convs = torch.nn.ModuleList()
+        self.convs.append(GATConv(in_channels, hidden_channels, heads))
+        for _ in range(num_layers - 2):
+            self.convs.append(
+                GATConv(heads * hidden_channels, hidden_channels, heads))
+        self.convs.append(
+            GATConv(heads * hidden_channels, out_channels, heads,
+                    concat=False))
+
+        self.skips = torch.nn.ModuleList()
+        self.skips.append(Lin(in_channels, hidden_channels * heads))
+        for _ in range(num_layers - 2):
+            self.skips.append(
+                Lin(hidden_channels * heads, hidden_channels * heads))
+        self.skips.append(Lin(hidden_channels * heads, out_channels))
+        self.dropout = dropout
+
+    def reset_parameters(self):
+        for conv in self.convs:
+            conv.reset_parameters()
+        for skip in self.skips:
+            skip.reset_parameters()
+
+    def forward(self, x, edge_index):
+        for i, (conv, skip) in enumerate(zip(self.convs, self.skips)):
+            x = conv(x, edge_index) + skip(x)
+            if i != self.num_layers - 1:
+                x = F.elu(x)
+                x = F.dropout(x, p=self.dropout, training=self.training)
+        return x
+
+    def inference(self, x_all):
+        pbar = tqdm(total=x_all.size(0) * self.num_layers)
+        pbar.set_description('Evaluating')
+
+        # Compute representations of nodes layer by layer, using *all*
+        # available edges. This leads to faster computation in contrast to
+        # immediately computing the final representations of each batch.
+        for i in range(self.num_layers):
+            xs = []
+            for batch in subgraph_loader:
+                x = x_all[batch.n_id].to(device)
+                edge_index = batch.edge_index.to(device)
+                x = self.convs[i](x, edge_index) + self.skips[i](x)
+                x = x[:batch.batch_size]
+                if i != self.num_layers - 1:
+                    x = F.elu(x)
+                xs.append(x.cpu())
+
+                pbar.update(batch.batch_size)
+
+            x_all = torch.cat(xs, dim=0)
+
+        pbar.close()
+
+        return x_all
+
+
+if args.use_gat:
+    print("GAT model")
+    model = GAT(dataset.num_features, num_hidden_channels, dataset.num_classes,
+                num_layers=num_layers, dropout=args.dropout,
+                heads=args.num_heads)
+else:
+    print("GraphSage model")
+    model = SAGE(dataset.num_features, num_hidden_channels,
+                 dataset.num_classes, num_layers=num_layers,
+                 dropout=args.dropout)
+
+model = GAT(dataset.num_features, num_hidden_channels, dataset.num_classes,
+            num_layers=num_layers, dropout=args.dropout,
+            heads=args.num_heads).to(device)
```

Comment:
The script never uses `SAGE`

Reply:
Default model used for training is SAGE model. If the user would like to test GAT model, can use --use_gat